### PR TITLE
add flex grid

### PR DESCRIPTION
**Pull request by mikeyrayvon**

### CODE REVIEW

File: css/grid.styl [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+.justify-end
+  justify-content: flex-end
+  
+.justify-between
+  justify-content: space-between
+  
+.justify-around
+  justify-content: space-around
+  
+$golden-ratio = 1.61803398875
```
Comment (varffs) [inline css/grid.styl]: we dont need the hard value cus we got kouto swiss. We can use the golden ratio scaler or something smaller from here: http://kouto-swiss.io/docs.html#modular-scale. `perfect-fourth` sounds like it might be nice

File: css/grid.styl [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+  for col in 1..$cols
+    .item-{$size}-{col}
+      flex-basis: ( 100% / $cols ) * col
+    .grid-row
+      .item-{$size}-{col}
+        max-width: ( 100% / $cols ) * col
+    .grid-column
+      .item-{$size}-{col}
+        max-height: ( 100% / $cols ) * col
```
Comment (varffs) [inline css/grid.styl]: is this def wanted? will make the grid work the same vertically. i.e. a 12 col but based on percent height of the `.grid-column`

File: css/site.styl [hunk 3]
```diff
@@ -124,16 +132,22 @@ img
 
 // MEDIA QUERIES
 
+grid(96%, s, 18px)
+
 @import "responsive/mobile"
 
 @media screen and (min-width: 720px)
 
+  grid(720px, m, 24px)
```
Comment (varffs) [inline css/site.styl]: do we normally set the container a bit smaller than the breakpoint/fullscreen?

File: css/site.styl [hunk 4]
```diff
@@ -124,16 +132,22 @@ img
...
+
   @import "responsive/tablet"
 
 // @media screen and (min-width: 1024px)
```
Comment (varffs) [inline css/site.styl]: should we do a grid for all these common screen sizes? means a little more to check as dev but will make the site more "responsive"

File: css/grid.styl [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+    flex-direction: column-reverse
+    
+.grid-item
+  position: relative
+  flex: 0 1 auto
+  box-sizing: border-box
+  max-width: 100%
+  max-height: 100%
+
+.no-gutter
```
Comment (varffs) [inline css/grid.styl]: I think this needs to be `.grid-item.no-gutter` to override everywhere

File: css/grid.styl [hunk 6]
```diff
@@ -0,0 +1,227 @@
+// GRID
+
+.container
+  margin: 0 auto
+  max-width: 2400px
```
Comment (mikeyrayvon) [inline css/grid.styl]: perhaps this isn't necessary? smells bad ;)

File: css/site.styl (hunk 4 above)
Comment (mikeyrayvon) [inline css/site.styl]: yea lets do that

File: css/grid.styl (hunk 2 above)
Comment (mikeyrayvon) [inline css/grid.styl]: hm. im not sure actually

Comment (mikeyrayvon) [inline css/grid.styl]: i'll do a test page in this branch?

**Comment by mixcas** [comment]
@mikeyrayvon a test page would be perfect. As we discussed, I think the best approach is to have a `test-page.php` that WP could use as template for a *Test* page.

what do u think?

File: css/grid.styl (hunk 6 above)
Comment (varffs) [inline css/grid.styl]: i think it smells good

Comment (mikeyrayvon) [inline css/grid.styl]: good argument

Comment (varffs) [inline css/grid.styl]: well actually you are correct i didnt think about it properly. there will always be an explicit highest value for container from the media queries so this will never actually do anything

**Comment by mikeyrayvon** [comment]
ok im working on `page-test.php`

**Comment by mikeyrayvon** [comment]
actually i'll do a page template called `grid-test.php`, and we select it from dropdown

File: css/site.styl [hunk 7]
```diff
@@ -124,15 +132,19 @@ img
...
 @import "responsive/mobile"
 
 @media screen and (min-width: 720px)
 
+  grid(680px, m, 21px)
+
   @import "responsive/tablet"
 
-// @media screen and (min-width: 1024px)
+@media screen and (min-width: 1200px)
```
Comment (varffs) [inline css/site.styl]: why did you delete the 1024px size breakpoint :/

**Comment by varffs** [comment]
Im happy for the merge. @mixcas can you ok?
